### ISSUE TITLE
large area edit cursor disappears in certain positions in zoomed-out view

### ISSUE THREAD

**Issue by cheweytoo** [OpenRCT2/OpenRCT2#22208]
### Operating System

Ubuntu Linux 22.04.4

### OpenRCT2 build

a0f5d09 develop Linux AppImage

### Base game

RollerCoaster Tycoon 2

### Describe the issue

When zoomed-out to the maximum, large area editing cursors (terrain, water, building rights, object deletion…) disappear in certain positions:

https://github.com/OpenRCT2/OpenRCT2/assets/29800630/2972a074-7742-46c8-81fc-51edacb12c8b

If a cursor is in such a disappeared position, a click does nothing.

This happens in all render modes, and all window scale factors at least up to 2.5

### Steps to reproduce

- zoom out to max zoomed-out view
- enable a larger area editing cursor
- move it around the park, it will disappear in some positions

**Comment by mrmbernardi** [comment]
I think the problem is with this function:

https://github.com/OpenRCT2/OpenRCT2/blob/58e576688c57694b2d7192575f354e7ae0630db0/src/openrct2/interface/Viewport.cpp#L1695-L1835

It's ostensibly doing something simple although the code is really verbose and hard to follow. If I get rid of the rounding adjustments by setting `round` to 1, then the tool always reaches the ground. Although maybe its less accurate?

I'd be interested in knowing if the original game had this bug too, as I think this code is still largely the same as it was then. There's also another bug when you zoom in. It's possible to position the tool between sprites and hit none of them.

I think the easiest thing might be just to rewrite all the code here from scratch.

**Comment by J-PIE-314** [comment]
This disappearance behavior is present in vanilla RCT2, making this an original bug.
However, the effect of clicking while the selection is invisible differs; in vanilla RCT2 the terrain selection will not be fixed in place and will continuously raise or lower whatever tiles you drag the cursor over.

**Comment by cheweytoo** [comment]
Oh crud, it still happens on some slope edges:


https://github.com/user-attachments/assets/55835950-eb70-4baa-8094-a0689b0af50f

**Comment by mrmbernardi** [comment]
I have a hunch that it's caused by an inconsistency in the way sprites are scaled down when zooming out, and that this is behind a lot of other issues too.

I'm not sure when I'll get around to addressing it so please make an issue.
